### PR TITLE
Remove fail-starting-node-on-ip-mismatch flag

### PR DESCRIPTION
**Pull request by freva**
Enabled everywhere since yesterday, 0 issues.

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java [hunk 1]
```diff
@@ -65,9 +65,10 @@ public GroupPreparer(NodeRepository nodeRepository, Optional<HostProvisioner> ho
         boolean dynamicProvisioningEnabled = hostProvisioner.isPresent() && dynamicProvisioningEnabledFlag
                 .with(FetchVector.Dimension.APPLICATION_ID, application.serializedForm())
                 .value();
+        // Do not in-place resize in dynamically provisioned zones
         boolean inPlaceResizeEnabled = enableInPlaceResize
                 .with(FetchVector.Dimension.APPLICATION_ID, application.serializedForm())
-                .value();
+                .value() && !dynamicProvisioningEnabled;
```
Comment (hakonhall) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java]: Unrelated change?

Comment (freva) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/provisioning/GroupPreparer.java]: Yes, but I too lazy to make a separate PR for that

**Review by hakonhall — APPROVED** [review]
